WebApp/autoreduce_webapp/utilities/pagination.py

```diff
@@ -0,0 +1,156 @@
+                    current_page = RunPage(next_page_index, self.items_per_page, True)
+                else:
+                    current_page = RunPage(next_page_index, self.items_per_page, False)
+                # Make sure we add the record to the new page we created
+                current_page.add_record(record)
+        current_page.set_start_and_end()
+        self.page_list.append(current_page)
+
+    def _create_display_list(self):
+        """
+        A compact version of the pagination table that is used for display purposes
+        """
+        for page in self.page_list:
+            if page.is_visible:
+                self.display_list.append(page)
+            else:
+                if len(self.display_list) >= 1 and self.display_list[-1] == "...":
+                    continue
+                else:
+                    self.display_list.append("...")
+
+    def _set_next_and_previous(self):
+        """
+        Sets class variables to describe what page is before and after the current page
+        """
+        if self.current_page_index < len(self.page_list):
+            self.next_page_index = self.current_page_index + 1
+            self.has_next = True
+        if self.current_page_index > 1:
+            self.previous_page_index = self.current_page_index - 1
+            self.has_previous = True
+
+
+# pylint:disable=too-few-public-methods
+class CustomPage(object):
+    """
+    Generic CustomPage class to model a single page in the pagination
+    """
+
+    def __init__(self, number, max_limit, is_visible):
+        """
+        :param number: The page number
+        :param max_limit: The most amount of data items allowed on a page
+        :param is_visible: Should this page be shown in the pagination?
+        """
+        self.records = []
+        self.is_visible = is_visible
+        self.number = number
+        self.max_limit = max_limit
+        self.display_name = None
+        self.start = 0
+        self.end = 0
+
+    def set_start_and_end(self):
+        """ Implemented by child class """
+        pass
+
+    def add_record(self, record):
+        """
+        Adds a record to the page, if the page has reached the limit that it can hold,
+        set the start and end run number variables and throw an exception to be caught
+        """
+        if len(self.records) >= self.max_limit:
+            self.set_start_and_end()
+            raise PageLimitException("Unable to add record to page. Max number of records reached")
+        self.records.append(record)
+
+
+# pylint:disable=too-few-public-methods
+class RunPage(CustomPage):
+    """
+    Specific implementation for rendering the pagination if sorting by Run Number
+    """
+
+    def set_start_and_end(self):
+        """
+        Custom function to set the start and end variables and
+        construct the display name for the page
+        """
+        self.start = self.records[0].run_number
+        self.end = self.records[-1].run_number
```

Comment:
I like that runs are now sorted by run number instead of date, but reruns are printed in reverse order, with the latest rerun appearing at the bottom. Is there a way that reruns can be sorted such that the latest appears first?

Reply:
This was actually put in by mistake. I agree that it is nicer, however we still need the date option as this is important for re-runs. e.g. if I re-run a run I expect it to be at the top of the page. 
I'll make the above amendment secondary to sort by version number (ascending), but I'll also add an option to sort by date too